SmartTutor/ContentModel/LearningObjects/ChallengeModel/FulfillmentStrategy/BasicMetricsChecker.cs

```diff
@@ -29,20 +33,49 @@ private List<CaDETMember> GetMethodsFromClasses(List<CaDETClass> caDETClasses)
 
         private bool ValidateClassMetricRules(List<CaDETClass> caDETClasses)
         {
+            int flag = 0;
+            _metricHints = new List<MetricHint>();
             foreach (CaDETClass caDETClass in caDETClasses)
+            {
                 foreach (MetricRangeRule classMetricRule in _classMetricRules)
+                {
                     if (!classMetricRule.MetricMeetsRequirements(caDETClass.Metrics))
-                        return false;
-            return true;
+                    {
+                        flag++;
+                        AddHintForUnfulfilledMetricRule(classMetricRule);
+                    }
+                }
+            }
+            if (flag == 0)
+                return true;
```

Comment:
First, you can replace this code with "return flag == 0".

Second, I'm not sure we should be returning booleans anymore. Since we want to highlight specific hints, maybe we can return a ChallengeValidationResult object that contains the status of the challenge (e.g., Success, Fail) and a list of hints that are relevant for the solutionAttempt.

Reply:
Yes, I agree. That is put as TODO and is next step for this PR.